devnotes/pnor_layout.md

```diff
@@ -0,0 +1,161 @@
+    uint32_t size;          // @size:       Partition size (in hdr.block_size)
+    uint32_t pid;           // @pid:        Parent partition entry (0xFFFFFFFF for toplevel)
+    uint32_t id;            // @id:         Partition entry ID [1..65536]
+    uint32_t type;          // @type:       Describe type of partition
+    uint32_t flags;         // @flags:      Partition attributes (optional)
+    uint32_t actual;        // @actual:     Actual partition size (in bytes)
+    uint32_t resvd[4];      // @resvd:      Reserved words for future use
+    struct {
+        uint32_t data[16];  // @user:       User data (optional)
+    } user;
+    uint32_t checksum;      // @checksum:   Partition entry checksum (includes all above)
+} __attribute__ ((packed));
+```
+
+Each partiton can contain image of following types:
+- FFS_TYPE_DATA
+- FFS_TYPE_LOGICAL
+- FFS_TYPE_PARTITION
+
+# Partition layout
+Data presented below was extracted using `./fcp -L file:flash.pnor -o 0` and
+enchanced with additional data extracted directly from `flash.pnor`.
+More on how to use this tool can be found in [porting.md#ffs-tools](porting.md#ffs-tools).
+
+Partition Table
+Version: 1
+Size: 2 * 0x1000 (block size)
+Block Count: 0x4000
+Entry Size: 0x80
+Number of entries: 0x21
+
+| Name                        | ID | Offset:End         | Size    | Type | Flags | Actual size  | coreboot usage              |
+| --------------------------- | -- | ------------------ | -------:| ---- | ----- | ------------:| --------------------------- |
+| [part](#part)               | 01 | 00000000:00001FFF  | 2000    | p    | p     | same as size | partition table             |
+| [HBEL](#HBEL)               | 02 | 00008000:0002BFFF  | 24000   | d    | -     | same as size | _unknown_                   |
+| [GUARD](#GUARD)             | 03 | 0002C000:00030FFF  | 5000    | d    | -     | same as size | _unknown_                   |
+| NVRAM                       | 04 | 00031000:000C0FFF  | 90000   | d    | -     | same as size | _unknown_                   |
+| [SECBOOT](#SECBOOT)         | 05 | 000C1000:000E4FFF  | 24000   | d    | -     | same as size | _unknown_                   |
+| DJVPD                       | 06 | 000E5000:0012CFFF  | 48000   | d    | -     | same as size | _unknown_                   |
+| MVPD                        | 07 | 0012D000:001BCFFF  | 90000   | d    | -     | same as size | _unknown_                   |
+| CVPD                        | 08 | 001BD000:00204FFF  | 48000   | d    | -     | same as size | _unknown_                   |
+| [HBB](#HBB)                 | 09 | 00205000:00304FFF  | 100000  | d    | -     | same as size | bootblock                   |
+| HBD                         | 0A | 00305000:00424FFF  | 120000  | d    | -     | same as size | _unknown_                   |
+| [HBI](#HBI)                 | 0B | 00425000:019C4FFF  | 15A0000 | d    | -     | same as size | romstage, ramstage, payload |
```

Comment:
Maybe be explicit that payload is skiboot: `payload (skiboot)`.

Reply:
Fixed